actors/builtin/miner/miner_actor.go

```diff
@@ -734,20 +703,16 @@ func (a Actor) checkPoStProvingPeriodExpiration(rt Runtime) {
 
 	var st State
 	expired := rt.State().Transaction(&st, func() interface{} {
-		if !st.PoStState.isChallenged() {
-			return false // Already exited challenged state successfully prior to expiry.
-		}
 
-		window := power.SurprisePostChallengeDuration
-		if rt.CurrEpoch() < st.PoStState.SurpriseChallengeEpoch+window {
-			// Challenge not yet expired.
-			return false
+		window := power.FallbackPostChallengeDuration
+		if rt.CurrEpoch() < st.PoStState.ProvingPeriodStart+ProvingPeriod+window {
```

Comment:
This method would be better named `checkPoStWindowExpiration`.

This should not add ProvingPeriod as well as the window, just the start + window. I think the expected cases are:
- `now == start + window` (the window has expired, return true)
- `now < start` (a PoSt was submitted and the period start moved into the future)

So error case would be:
- `now > start && now != start+window` (cron mis-timed)

At present, this closure never returns `false`, but it should.